### PR TITLE
Jwt support

### PR DESCRIPTION
**Pull request by VlasenkoYevhenii**

### CODE REVIEW

**Review by karma-o — CHANGES_REQUESTED** [review]
Please change the base of this PR from `main` to the previous task branch and fix merge conflicts before sending to review

File: src/main/java/com/example/bookstoreapplication/controller/AuthController.java [hunk 1]
```diff
@@ -20,14 +23,23 @@
...
 public class AuthController {
-    private final UserService service;
+    private final UserService userService;
+    private final AuthenticationService authenticationService;
+
+    @PostMapping("/login")
+    @ResponseStatus(HttpStatus.ACCEPTED)
+    @Operation(summary = "Login endpoint",
+                description = "Returns JWT token in response")
+    public UserLoginResponseDto login(@RequestBody UserLoginRequestDto dto) {
```
Comment (d1sam) [inline src/main/java/com/example/bookstoreapplication/controller/AuthController.java:33]:
```suggestion
    public UserLoginResponseDto login(@RequestBody UserLoginRequestDto request) {
```

File: src/main/java/com/example/bookstoreapplication/dto/user/UserLoginRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import org.hibernate.validator.constraints.Length;
+
+@Data
+public class UserLoginRequestDto {
+    @NotBlank(message = "The email field is empty")
+    @Email(message = "The email format is invalid")
+    private String email;
+
+    @NotBlank(message = "The password field is empty")
+    @Length(min = 6)
```
Comment (d1sam) [inline src/main/java/com/example/bookstoreapplication/dto/user/UserLoginRequestDto.java:15]: what about max?

File: src/main/java/com/example/bookstoreapplication/dto/user/UserLoginRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotBlank;
+import lombok.Data;
+import org.hibernate.validator.constraints.Length;
+
+@Data
+public class UserLoginRequestDto {
+    @NotBlank(message = "The email field is empty")
+    @Email(message = "The email format is invalid")
+    private String email;
```
Comment (d1sam) [inline src/main/java/com/example/bookstoreapplication/dto/user/UserLoginRequestDto.java:12]: what about the size?

File: src/main/resources/application.properties
Comment (d1sam) [inline src/main/resources/application.properties:1]: let's remove secret value and use from env properties

File: src/test/resources/application.properties
Comment (d1sam) [inline src/test/resources/application.properties:1]: let's remove secret value and use from env properties

**Review by d1sam — CHANGES_REQUESTED** [review]
Good job, but there are some minor changes, that should be done according my comments.

File: src/test/resources/application.properties
Comment (VlasenkoYevhenii) [reply src/test/resources/application.properties:1]: Changing prop to env prop in test folder causes CI failure

File: src/main/java/com/example/bookstoreapplication/dto/user/UserLoginRequestDto.java (hunk 3 above)
Comment (andrii-hoienko) [reply src/main/java/com/example/bookstoreapplication/dto/user/UserLoginRequestDto.java:12]: Not resolved

File: src/main/java/com/example/bookstoreapplication/security/JwtAuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+                    userDetails.getPassword(), userDetails.getAuthorities());
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader(AUTHORIZATION_HEADER);
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith(BEARER)) {
+            return bearerToken.substring(TOKEN_START_INDEX);
```
Comment (andrii-hoienko) [inline src/main/java/com/example/bookstoreapplication/security/JwtAuthenticationFilter.java:48]:
```suggestion
            return bearerToken.substring(BEARER.length);
```

File: src/main/java/com/example/bookstoreapplication/security/JwtAuthenticationFilter.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@RequiredArgsConstructor
+@Component
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+    private static final int TOKEN_START_INDEX = 7;
```
Comment (andrii-hoienko) [inline src/main/java/com/example/bookstoreapplication/security/JwtAuthenticationFilter.java:21]:
```suggestion
```

File: src/main/resources/application.properties [hunk 6]
```diff
@@ -7,3 +7,7 @@ spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
 spring.jpa.hibernate.ddl-auto=validate
 spring.jpa.show-sql=true
 spring.jpa.open-in-view=false
+
+jwt.expiration=300000
+jwt.secret=${JWT_SECRET}
+
```
Comment (andrii-hoienko) [inline src/main/resources/application.properties:13]:
```suggestion
```

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Nice, Left some comments

**Review by andrii-hoienko — APPROVED** [review]
Good job!

**Pull request merged**